core/src/main/java/org/mule/runtime/core/api/diagnostics/notification/ProfilerNotificationListener.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.core.api.diagnostics.notification;
+
+import org.mule.api.annotation.Experimental;
+import org.mule.runtime.api.notification.NotificationListener;
+
+/**
+ * An interface for {@link ProfilingNotification} listeners.
+ *
+ * @param <T> extension of {@link ProfilingNotification}
+ */
+@Experimental
+public interface ProfilerNotificationListener<T extends ProfilingNotification>
```

Comment:
shouldn't this go into mule-api? o a new api module for using this functionality?

Reply:
Moved to mule-api. If this results in a more decoupling component after the experimental phase is finished, we'll move this to a different module. For the time being, the beans associated to the discovery service and profiling service will be  implemented and managed in the runtime.